### PR TITLE
Initial MVP

### PR DESCRIPTION
**Pull request by danielbachhuber**

### CODE REVIEW

File: script.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+def parse_videos_from_feed():
+    """
+    Injest MRSS feed into local scope
+    """
+
+    data = feedparser.parse(os.getenv('MTFV_MRSS_URL'))
+    videos = []
+    for i, video in enumerate(data.entries):
+        videos.append( video )
```
Comment (danielbachhuber) [inline script.py]: @scribu in this method, I'd like to dedupe videos that have already been uploaded into Facebook. Thoughts on the best way to do this?

Couple of ideas:

1. Store the original filename in a Redis instance when it's uploaded, and dedupe on that. However, if we lose the Redis data, we run the risk of uploading everything again.
1. Store a `last_run_date` somewhere, and only upload videos published after that date.

Comment (scribu) [inline script.py]: I'd go with Redis.

With option 2, you run the risk of not uploading certain videos, if a bug occurs.

Comment (scribu) [inline script.py]: Besides, the `last_run_date` would also need to be stored in Redis, since Heroku doesn't persist any files written to the filesystem.

**Comment by danielbachhuber** [comment]
@mpatek #reviewmerge?

File: script.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+
+def parse_videos_from_feed():
+    """
+    Injest MRSS feed into local scope; format videos to FB upload spec
+    """
+
+    data = feedparser.parse(os.getenv('MTFV_MRSS_URL'))
+    videos = []
+    for i, video in enumerate(data.entries):
```
Comment (mpatek) [inline script.py]: List comprehensions are a cool feature of Python:

```Python
return [{
        'title': video['title'],
        'description': video['summary'],
        'file_url': video['media_content'][0]['url'],
        'file_size': video['media_content'][0]['filesize']} for video in data.entries]
```

File: script.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        os.environ[k] = v
+
+def upload_video_to_facebook(video):
+    """
+    Uploads a given video to Facebook Graph API
+    """
+    from oauth import authorize_installed_app
+
+    http = authorize_installed_app(
+        scope=('publish_actions'),
```
Comment (mpatek) [inline script.py]: This should probably be: `scope=('publish_actions',)` so that Python knows it's a tuple.

File: script.py (hunk 2 above)
Comment (danielbachhuber) [inline script.py]: That would return an array?

File: script.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+
+@click.command()
+@click.argument('secrets_path', type=click.Path(exists=True))
+@click.argument('token_path', type=click.Path())
+@click.argument('scope', required=True, nargs=-1)
+def oauth(secrets_path, token_path, scope):
+    """Generate an OAuth token."""
+    from oauth import get_credentials
+    print secrets_path
```
Comment (mpatek) [inline script.py]: Take out this `print` statement?

File: script.py (hunk 3 above)
Comment (danielbachhuber) [inline script.py]: ac2aec5

File: script.py (hunk 4 above)
Comment (danielbachhuber) [inline script.py]: 4805766

File: script.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    get_credentials(scope, secrets_path, token_path)
+
+
+cli.add_command(oauth)
+
+
+@click.command()
+def upload():
+    """Upload videos to Facebook from MRSS feed"""
+    update_env();
```
Comment (mpatek) [inline script.py]: Maybe `update_env` should go at the beginning of `cli`?

File: script.py (hunk 2 above)
Comment (danielbachhuber) [inline script.py]: 5dba944

File: script.py (hunk 5 above)
Comment (danielbachhuber) [inline script.py]: Good suggestion, a6fd6c0

**Comment by danielbachhuber** [comment]
@mpatek I hit all of your suggestions.

File: .env.sample [hunk 6]
```diff
@@ -0,0 +1,2 @@
+MTFV_MRSS_URL=
+MTFV_FACEBOOK_ENTITY_ID=
```
Comment (mpatek) [inline .env.sample]: Maybe add `FACEBOOK_OAUTH=` here?

Comment (mpatek) [inline .env.sample]: ah nevermind. maybe better to include some instructions about how to store it.

File: script.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+            'handlers': ['console'],
+        }
+    })
+
+
+@click.command()
+@click.argument('secrets_path', type=click.Path(exists=True))
+@click.argument('token_path', type=click.Path())
+@click.argument('scope', required=True, nargs=-1)
+def oauth(secrets_path, token_path, scope):
```
Comment (mpatek) [inline script.py]: Is this command going to be used?

Comment (danielbachhuber) [inline script.py]: > Is this command going to be used?

Yep — this project is public. I'll add some instructions to README about setup.

**Pull request merged**